ansible-ipi-install/roles/node-prep/tasks/100_power_off_cluster_servers.yml

```diff
@@ -19,5 +19,6 @@
         password: "{{ hostvars[item]['ipmi_password'] }}"
         port: "{{ hostvars[item]['ipmi_port'] | default(623) }}"
         state: off
+      when: groups['poweroff_hosts'] is defined
```

Comment:
I'm not sure we need this because lines 9,10 take care of what hosts should be part of the poweroff_hosts group. The masters should realistically always been in that group. 

What scenario are you running that gave you an error?

Reply:
If all of your masters have poweroff=false, then the poweroff_hosts group never gets created and this task fails